src/types.jl

```diff
@@ -53,6 +53,13 @@ function breakpointchar(bps::BreakpointState)
     return bps.condition === falsecondition ? ' ' : 'd'     # no breakpoint : disabled
 end
 
+abstract type AbstractFrameInstance end
+mutable struct DispatchableMethod
+    next::Union{Nothing,DispatchableMethod}  # linked-list representation
+    frameinstance::Union{Compiled, AbstractFrameInstance} # really a Union{Compiled, FrameInstance} but we have a cyclic dependency
```

Comment:
This could hurt. Union-splitting may not work, unless the compiler is smart enough to know there is only one concrete subtype.

Reply:
Yeah but we type assert immediately when we use it anyway.

Reply:
Looked a bit, there is a single place where we look at this field and all uses either just calls `isa` on it, or type asserts before it is used so there should be no dynamic behavior.